From 789ee5adfa484de1da1e91edc254471441ed9947 Mon Sep 17 00:00:00 2001
From: Adriana Reus <adriana.reus@nxp.com>
Date: Mon, 28 Aug 2017 13:48:00 +0300
Subject: [PATCH 2463/5242] MLK-16281-1: clk-imx8qm: Remove duplicated gpt
 clocks

commit  777305f3d247938b96933a55f75cc04114df1382 from
https://source.codeaurora.org/external/imx/linux-imx.git

Some gpt clocks are defined twice which results in:

 gpt0_div
    gpt0_clk

and also:

 gpt_0_div
    gpt_0_ipg_s_clk
       gpt_0_ipg_slv_clk
          gpt_0_hf_clk
          gpt_0_clk

The second version is correct as per gpt lpcg cell.
This patch removes the first set of clocks.

Signed-off-by: Adriana Reus <adriana.reus@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8qm.c |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/clk/imx/clk-imx8qm.c b/drivers/clk/imx/clk-imx8qm.c
index f736dc9..7244da3 100644
--- a/drivers/clk/imx/clk-imx8qm.c
+++ b/drivers/clk/imx/clk-imx8qm.c
@@ -236,11 +236,6 @@ static int imx8qm_clk_probe(struct platform_device *pdev)
 	clks[IMX8QM_ENET1_RGMII_DIV] = imx_clk_divider_scu("enet1_rgmii_div", SC_R_ENET_1, SC_PM_CLK_MISC0);
 	clks[IMX8QM_GPMI_BCH_IO_DIV] = imx_clk_divider_scu("gpmi_io_div", SC_R_NAND, SC_PM_CLK_MST_BUS);
 	clks[IMX8QM_GPMI_BCH_DIV] = imx_clk_divider_scu("gpmi_bch_div", SC_R_NAND, SC_PM_CLK_PER);
-	clks[IMX8QM_GPT0_DIV] = imx_clk_divider_scu("gpt0_div", SC_R_GPT_0, SC_PM_CLK_PER);
-	clks[IMX8QM_GPT1_DIV] = imx_clk_divider_scu("gpt1_div", SC_R_GPT_1, SC_PM_CLK_PER);
-	clks[IMX8QM_GPT2_DIV] = imx_clk_divider_scu("gpt2_div", SC_R_GPT_2, SC_PM_CLK_PER);
-	clks[IMX8QM_GPT3_DIV] = imx_clk_divider_scu("gpt3_div", SC_R_GPT_3, SC_PM_CLK_PER);
-	clks[IMX8QM_GPT4_DIV] = imx_clk_divider_scu("gpt4_div", SC_R_GPT_4, SC_PM_CLK_PER);
 	clks[IMX8QM_USB3_ACLK_DIV] = imx_clk_divider_scu("usb3_aclk_div", SC_R_USB_2, SC_PM_CLK_PER);
 	clks[IMX8QM_USB3_BUS_DIV] = imx_clk_divider_scu("usb3_bus_div", SC_R_USB_2, SC_PM_CLK_MST_BUS);
 	clks[IMX8QM_USB3_LPM_DIV] = imx_clk_divider_scu("usb3_lpm_div", SC_R_USB_2, SC_PM_CLK_MISC);
@@ -341,11 +336,6 @@ static int imx8qm_clk_probe(struct platform_device *pdev)
 	clks[IMX8QM_SDHC0_CLK] = imx_clk_gate_scu("sdhc0_clk", "sdhc0_div", SC_R_SDHC_0, SC_PM_CLK_PER, (void __iomem *)(USDHC_0_LPCG), 0, 0);
 	clks[IMX8QM_SDHC1_CLK] = imx_clk_gate_scu("sdhc1_clk", "sdhc1_div", SC_R_SDHC_1, SC_PM_CLK_PER, (void __iomem *)(USDHC_1_LPCG), 0, 0);
 	clks[IMX8QM_SDHC2_CLK] = imx_clk_gate_scu("sdhc2_clk", "sdhc2_div", SC_R_SDHC_2, SC_PM_CLK_PER, (void __iomem *)(USDHC_2_LPCG), 0, 0);
-	clks[IMX8QM_GPT0_CLK] = imx_clk_gate_scu("gpt0_clk", "gpt0_div", SC_R_GPT_0, SC_PM_CLK_PER, (void __iomem *)(GPT_0_LPCG + 0x4), 0, 0);
-	clks[IMX8QM_GPT1_CLK] = imx_clk_gate_scu("gpt1_clk", "gpt1_div", SC_R_GPT_1, SC_PM_CLK_PER, (void __iomem *)(GPT_1_LPCG + 0x4), 0, 0);
-	clks[IMX8QM_GPT2_CLK] = imx_clk_gate_scu("gpt2_clk", "gpt2_div", SC_R_GPT_2, SC_PM_CLK_PER, (void __iomem *)(GPT_2_LPCG + 0x4), 0, 0);
-	clks[IMX8QM_GPT3_CLK] = imx_clk_gate_scu("gpt3_clk", "gpt3_div", SC_R_GPT_3, SC_PM_CLK_PER, (void __iomem *)(GPT_3_LPCG + 0x4), 0, 0);
-	clks[IMX8QM_GPT4_CLK] = imx_clk_gate_scu("gpt4_clk", "gpt4_div", SC_R_GPT_4, SC_PM_CLK_PER, (void __iomem *)(GPT_4_LPCG + 0x4), 0, 0);
 	clks[IMX8QM_ENET0_AHB_CLK]   = imx_clk_gate2_scu("enet0_ahb_clk", "axi_conn_clk_root", (void __iomem *)(ENET_0_LPCG), 8, FUNCTION_NAME(PD_CONN_ENET_0));
 	clks[IMX8QM_ENET0_IPG_S_CLK] = imx_clk_gate2_scu("enet0_ipg_s_clk", "ipg_conn_clk_root", (void __iomem *)(ENET_0_LPCG), 20, FUNCTION_NAME(PD_CONN_ENET_0));
 	clks[IMX8QM_ENET0_IPG_CLK]   = imx_clk_gate2_scu("enet0_ipg_clk", "enet0_ipg_s_clk", (void __iomem *)(ENET_0_LPCG), 16, FUNCTION_NAME(PD_CONN_ENET_0));
-- 
1.7.9.5

